src/main/kotlin/ar/edu/unahur/obj2/socios/Cliente.kt

```diff
@@ -1,4 +1,16 @@
 package ar.edu.unahur.obj2.socios
 
-class Cliente {
+class Cliente(private var dineroEnBolsillo : Double, private var estadoDeAnimo: EstadosDeAnimo, private var barrio: Barrio) {
+
+    private var importePedido : Double = 0.0
+
+    fun realizarPedido(importe : Double) {
+        this.importePedido = importe
+    }
+
+    fun darPropina(): Double {
+        val propina = estadoDeAnimo.aplicarEstadoAnimo(importePedido, dineroEnBolsillo)
+        return barrio.aplicarPropinaPorBarrio(propina)
+    }
```

Comment:
No es necesario guardar `propina`, en todo caso estaría mejor usar `val` en vez de `var`.
```suggestion
    fun darPropina() = barrio.aplicarPropinaPorBarrio(estadoDeAnimo.aplicarEstadoAnimo(importePedido, dineroEnBolsillo))

```